sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2106,13 +2106,19 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
-  val LEGACY_CTE_PRECEDENCE_ENABLED = buildConf("spark.sql.legacy.ctePrecedence.enabled")
+  object LegacyBehaviorPolicy extends Enumeration {
+    val EXCEPTION, LEGACY, NEW_BEHAVIOR = Value
```

Comment:
`NEW_BEHAVIOR` sounds not a good name. `INNERFIRST`?

Reply:
Or `INNER_PRECEDENCE`?

Reply:
Thanks for the advice @viirya. Yes, both `INNER_FIRST` and `INNER_PRECEDENCE` are more accurate in this PR, but I also want to make this enum as a common config value, which can be reused in other similar issues. WDYT?

Reply:
how about `CORRECTED` to indicate that it's the corrected behavior?

Reply:
ok.